flink-connector-kafka/pom.xml

```diff
@@ -72,6 +72,13 @@ under the License.
             <version>${kafka.version}</version>
         </dependency>
 
+        <!-- When flink-core bump to commons-collections4, we can remove it.-->
```

Comment:
With the externalization of connectors, we shouldn't have the connector depend on an artifact that may or may not be bundled with Flink.

Reply:
@MartijnVisser it will be removed when flink update finished. Now it'a circular.

flink-python test case in flink reference flink-sql-connector-kafka（not uber jar, without commons-collections class）, so search it in flink and cause ClassNotFoundException. 

And this kafka connector also reference flink-core. 

Must break circular references from side. Is there a better way？

Reply:
Another way is flink keep commons-collections 3.x remained and add commons-collections4.
then kafka connector in this repo update to commons-collections4, and flink remove commons-collections 3.x.  it's flink-side way. what do u think?

Reply:
> Another way is flink keep commons-collections 3.x remained and add commons-collections4. then kafka connector in this repo update to commons-collections4 (just replace package name, no dependency import), and flink remove commons-collections 3.x. it's [flink-side](https://issues.apache.org/jira/browse/FLINK-side) solution. what do u think?

@MartijnVisser Whether we can keep these 2 version in flink temporarily? 


In fact, spark also encounters similar problems (but not reference loop), and currently two versions coexist.


https://github.com/apache/spark/pull/35257
https://github.com/apache/spark/pull/36009/files